src/Compilers/Core/Portable/MetadataReference/Metadata.cs

```diff
@@ -14,6 +14,11 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public sealed class MetadataId
     {
+        private MetadataId()
+        {
+        }
+
+        internal static MetadataId CreateNewId() => new MetadataId();
```

Comment:
But does the compiler have other IDs it creates?  What's the pattern is uses for those?